akka-remote/src/main/scala/akka/remote/AddressUidExtension.scala

```diff
@@ -22,15 +24,18 @@ object AddressUidExtension extends ExtensionId[AddressUidExtension] with Extensi
   override def lookup = AddressUidExtension
 
   override def createExtension(system: ExtendedActorSystem): AddressUidExtension = new AddressUidExtension(system)
+
 }
 
 class AddressUidExtension(val system: ExtendedActorSystem) extends Extension {
-  val longAddressUid: Long = {
-    // FIXME we should use a long here, but then we need to change in Cluster and RemoteWatcher also
-    //ThreadLocalRandom.current.nextLong()
-    ThreadLocalRandom.current.nextInt()
-  }
 
+  val longAddressUid: Long = ThreadLocalRandom.current.nextLong()
+
+  // used by old remoting and part of public api
   @deprecated("Use longAddressUid instead", "2.4.x")
-  val addressUid: Int = longAddressUid.toInt
+  lazy val addressUid: Int = {
```

Comment:
It's still just an accessor method, even if it is lazy, so shouldn't be a problem.